docs/docs/data-fetching.md

```diff
@@ -0,0 +1,169 @@
+
+Source plugins take advantage of the [`sourceNodes` API](/docs/node-apis/#sourceNodes) and the [`createNode` action](/docs/actions/#createNode) provided by Gatsby to make your data queryable during the build process.
+
+#### Writing a query to gather the static data needed for a page
+
+With the source plugin installed and added to your config, you can write a [static query](/docs/use-static-query/) that will pull the data from GraphQL API and will retrieve the necessary data while building the site.
+
+```jsx:title=src/pages/index.js
+import React from "react"
+import { graphql, useStaticQuery } from "gatsby" // highlight-line
+
+const IndexPage = () => {
+  // highlight-start
+  const gatsbyRepoData = useStaticQuery(graphql`
+    query {
+      github {
+        repository(name: "gatsby", owner: "gatsbyjs") {
+          id
+          nameWithOwner
+          url
+        }
+      }
+    }
+  `)
+  // highlight-end
+
+  return (
+    <section>
+      <p>
+        Build Time Data: Gatsby repo{` `}
+        <a href="gatsbyRepoData.github.repository.nameWithOwner.url">
+          {gatsbyRepoData.github.repository.nameWithOwner} // highlight-line
+        </a>
+      </p>
+    </section>
+  )
+}
+
+export default IndexPage
+```
+
+### Fetching data at runtime
+
+For fetching data at runtime, you can use any method to retrieve data that you would use in a regular React app.
+
+#### Retrieving data with the `fetch` API
+
+The `fetch` API is a modern implementation of the older, well-supported `XMLHttpRequest`.
+
+With the `useState` and `useEffect` hooks from React, you can query for the data once when the page loads, and save the data returned to a variable called `starsCount`. Every time the page is refreshed, `fetch` will go to the GitHub API to retrieve the most up-to-date version of the data.
+
+```jsx:title=src/pages/index.js
+import React, { useState, useEffect } from "react" // highlight-line
+import { graphql, useStaticQuery } from "gatsby"
+
+const IndexPage = () => {
+  const gatsbyRepoData = useStaticQuery(graphql`
+    query {
+      github {
+        repository(name: "gatsby", owner: "gatsbyjs") {
+          id
+          nameWithOwner
+          url
+        }
+      }
+    }
+  `)
+  // highlight-start
+  const [starsCount, setStarsCount] = useState(0)
+  useEffect(() => {
+    const fetchData = async () => {
+      // get data from GitHub api
+      const result = await fetch(`https://api.github.com/repos/gatsbyjs/gatsby`)
+      // parse JSON from request
+      const resultData = await result.json()
+      // set data for the number of stars
+      setStarsCount(resultData.stargazers_count)
+    }
+    fetchData()
+  }, [])
+  // highlight-end
+
+  return (
+    <section>
+      <p>
+        Build Time Data: Gatsby repo{` `}
```

Comment:
What is the ``{`  `}`` for here, out of curiosity?

Reply:
When prettier wraps the text for readability (or whatever reason they wrap text) it pushes new HTML elements like `<a>` down a line and will put text inside an anchor next to the text inside the `<p>` tag. Adding `{` `}` adds a space, sometimes prettier even automatically adds it.